pallets/collator-assignment/src/assignment.rs

```diff
@@ -435,7 +438,9 @@ where
         let mut needed_new_collators = 0;
         for (para_id, num_collators) in chains.iter() {
             let cs = old_assigned.entry(*para_id).or_default();
-            needed_new_collators += (*num_collators as usize).saturating_sub(cs.len());
+            needed_new_collators = needed_new_collators
```

Comment:
this can be saturating_accrue and saturating_reduce no?

Reply:
So  `saturating_accrue` and `saturating_reduce` and directly modifies `needed_new_collators`. So it needs to be applies either this way or  or  as 

```
needed_new_collator.saturating_accrue();  
needed_new_collator.saturating_reduce();
```

Otherwise it does not calculate the correct value and breaks the tests.

Reply:
ok then leave it as it is, im fine with it!